### DISCUSSION TITLE
Is this expected behavior for using a Dependency inside a Dependency?

### DISCUSSION THREAD

**Discussion by ghost** [pointfreeco/swift-dependencies#67] [Q&A] [answered]
I have a DependencyA that depends on DependencyB. This is what the implementation looks like:

```
extension DependencyA: DependencyKey {

  public static var liveValue: Self {
    @Dependency(\.dependencyB) var dependencyB

    return Self(dependencyB: dependencyB)
  }

}
```

then I override DependencyB to use the test value inside the start of the app:

```
var body: some Scene {
  WindowGroup {
    withDependencies {
      $0.dependencyB = .testValue
    } operation: {
      AppView(
        viewModel: AppViewModel()
      )
    } 
  }
}
```

However, when I access DependencyA later in the app, it seems like it's using the `liveValue` version of DependencyB. However if I access DependencyB here, it returns the `testValue` as expected. 

In my head, at this point DependencyA should be using the `testValue` of DependencyB. I'm wondering if this is the expected behavior or did I implement this incorrectly?

Regarding the architecture of the app, I'm using ViewModels instead of TCA. The app is still fairly small right now and the point where I access DependencyA and it's returning unexpected results is only 3 scenes into the app and I've made sure to always wrap creation of any `ViewModel` with `withDependencies(from: self)` with the viewModel being a `ObservedObject` that is injected into the scene. 

If I overwrite any dependencies it seems like it's being propogated correctly, the only case this doesn't seem to work is when I have a dependency depending on another dependency as listed above.

As a workaround, right now I just override both dependencies with `testValue` but I would like to only need to override DependencyB and have it reflected in DependencyA.

I've searched around and AFAIK this is the recommended way to accomplish what I'm trying to do. Any help is appreciated. Thanks!

**Comment by mbrandonw** [comment, marked as answer]
This may be related to #71 and we are currently trying to figure out the best way forward. It does seem that using dependencies from dependencies in non-"single entry point" applications comes with some subtle edge cases. We hope to improve the tools soon.

**Reply by ghost** [reply]
Hey thanks for the reply. That does seem like the same issue I'm facing. I'll keep an eye on it. Thanks!
